### PR TITLE
Walkaround bug in route controller.

### LINKED ISSUE #28362: After upgrading nodes they are marked as "NetworkUnavailable"
```
E0701 15:46:30.814395       5 routecontroller.go:130] Could not create route 019d3522-3fa2-11e6-a04f-42010af00002 10.180.3.0/24 for node e2e-test-filipg-minion-r6b8 after 209.159431ms: googleapi: Error 409: The resource 'projects/ivory-honor-865/global/routes/e2e-test-filipg-019d3522-3fa2-11e6-a04f-42010af00002' already exists, alreadyExists
E0701 15:46:30.816208       5 routecontroller.go:130] Could not create route 5e7093cb-3fa1-11e6-a04f-42010af00002 10.180.1.0/24 for node e2e-test-filipg-minion-apmw after 201.001883ms: googleapi: Error 409: The resource 'projects/ivory-honor-865/global/routes/e2e-test-filipg-5e7093cb-3fa1-11e6-a04f-42010af00002' already exists, alreadyExists
E0701 15:46:30.949416       5 routecontroller.go:130] Could not create route b978bdb7-3fa1-11e6-a04f-42010af00002 10.180.2.0/24 for node e2e-test-filipg-minion-nu8c after 227.942663ms: googleapi: Error 409: The resource 'projects/ivory-h
```

### PR DESCRIPTION
**Pull request by Random-Liu**
Fix https://github.com/kubernetes/kubernetes/issues/28362.

This PR didn't fix the original bug, because we still didn't understand it and even haven't successfully reproduced it yet. This PR only makes sure that route controller won't mark the node as NetworkUnavailable if the route is already created.

I don't know whether we want a walkaround like this, so I haven't added a unit test yet. If we want this, I'll add a unit test.

@dchen1107 @erictune @fgrzadkowski 

[Analytics]()

### CODE REVIEW

File: pkg/controller/route/routecontroller.go [hunk 1]
```diff
@@ -87,6 +88,12 @@ func (rc *RouteController) reconcileNodeRoutes() error {
 	return rc.reconcile(nodeList.Items, routeList)
 }
 
+const alreadyExistError = "googleapi: Error 409"
```
Comment (dchen1107) [inline pkg/controller/route/routecontroller.go]: I know this is a workaround for GCE. Just feel uncomfortable to have googleapi to this generic error: alreadyExistError.

**Comment by dchen1107** [comment]
@erictune This is the pr I mentioned to you earlier. Since we couldn't reproduce the issue, I decided not to fix for 1.3. But this is the workaround if we do believe there is an issue in 1.3 and need a cherrypick.

**Comment by dchen1107** [comment]
Can we have a unittest for this one?

**Comment by Random-Liu** [comment]
@dchen1107 Sure. I'll add a unit test.

**Comment by Random-Liu** [comment]
@dchen1107 @fgrzadkowski added unit test.

**Comment by dchen1107** [comment]
The failure is never occurred after the initial reports. The pr itself just walkaround this unproducible issue, I decided to close it without any change.

**Pull request closed without merging**